spyderlib/config/user.py

```diff
@@ -117,8 +118,27 @@ def _write_file(fname):
                 raise(e)
 
     def filename(self):
+        """Defines the name of the configuration file to use."""
+        self._filename = getattr(self, '_filename', None)
+        self._root_path = getattr(self, '_root_path', None)

```

Comment:
Why not? `UserConfig` inherits from `DefaultsConfig`, and `ProjectsConfig` inherits from `UserConfig`.

So if you add these attrs to `DefaultsConfig`'s `__init__`, they should be there for `ProjectsConfig`, right?